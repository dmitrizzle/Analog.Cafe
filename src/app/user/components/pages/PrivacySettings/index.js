import React from "react"

import ArticleSection from "../../../../core/components/pages/Article/components/ArticleSection"
import ArticleWrapper from "../../../../core/components/pages/Article/components/ArticleWrapper"
import Button from "../../../../core/components/controls/Button/components/Button"
import CardHeader from "../../../../core/components/controls/Card/components/CardHeader"
import CardIntegrated from "../../../../core/components/controls/Card/components/CardIntegrated"
import HeaderLarge from "../../../../core/components/vignettes/HeaderLarge"

export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      dataSharing: {
        googleAnalytics:
          localStorage.getItem("ga-enabled") === "false" ? false : true
      }
    }
  }
  handleToggleGA = () => {
    const googleAnalytics = !this.state.dataSharing.googleAnalytics
    this.setState({
      dataSharing: {
        ...this.state.dataSharing,
        googleAnalytics
      }
    })
    localStorage.setItem("ga-enabled", googleAnalytics)
  }
  render = () => {
    return (
      <ArticleWrapper>
        <HeaderLarge pageTitle="Privacy Settings" />
        <ArticleSection>
          <p>
            Google Analytics is used by Analog.Cafe to anonymously (individual
            users can not be identified) study how the visitors are using the
            website and how the website experience could be improved based on
            those findings.
          </p>
          <p>
            Analog.Cafe will store your preferences in your browserâ€™s{" "}
            <em>LocalStorage</em> (not a cookie) and remember them indefinitely,
            as long as your browser or you, personally, do not choose to remove
            that data.
          </p>
          <CardIntegrated>
            <CardHeader
              stubborn
              buttons={[0]}
              noStar
              title="Share Usage Data With:"
            />
            <Button
              onClick={this.handleToggleGA}
              inverse={this.state.dataSharing.googleAnalytics}
            >
              Google Analytics
            </Button>
          </CardIntegrated>
          <p style={{ textAlign: "center" }}>
            <small>
              Your usage data is
              {!this.state.dataSharing.googleAnalytics && (
                <strong> not</strong>
              )}{" "}
              being
              {this.state.dataSharing.googleAnalytics && (
                <span> anonymously</span>
              )}{" "}
              shared with Google Analytics.
            </small>
          </p>
        </ArticleSection>
      </ArticleWrapper>
    )
  }
}
