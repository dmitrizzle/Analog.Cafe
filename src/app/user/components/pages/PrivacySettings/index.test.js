import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import PrivacySettings from "./"

it("Render PrivacySettings, matches snapshot", () => {
  const element = shallow(<PrivacySettings />)
  expect(element).toMatchSnapshot()
})
