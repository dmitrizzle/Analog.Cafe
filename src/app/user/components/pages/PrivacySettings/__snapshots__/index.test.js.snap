// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivacySettings, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          pageTitle="Privacy Settings"
/>,
        <styled.section>
          <p>
                    Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.
          </p>
          <p>
                    Analog.Cafe will store your preferences in your browser’s
                     
                    <em>
                              LocalStorage
                    </em>
                     (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.
          </p>
          <Styled(styled.div)>
                    <WithTheme(Component)
                              buttons={
                                        Array [
                                                  0,
                                                ]
                              }
                              noStar={true}
                              stubborn={true}
                              title="Share Usage Data With:"
                    />
                    <Unknown
                              inverse={true}
                              loading={false}
                              onClick={[Function]}
                    >
                              Google Analytics
                    </Unknown>
          </Styled(styled.div)>
          <p
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          >
                    <small>
                              Your usage data is
                               
                              being
                              <span>
                                         anonymously
                              </span>
                               
                              shared with Google Analytics.
                    </small>
          </p>
</styled.section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pageTitle": "Privacy Settings",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <p>
              Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.
</p>,
            <p>
              Analog.Cafe will store your preferences in your browser’s
               
              <em>
                            LocalStorage
              </em>
               (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.
</p>,
            <Styled(styled.div)>
              <WithTheme(Component)
                            buttons={
                                          Array [
                                                        0,
                                                      ]
                            }
                            noStar={true}
                            stubborn={true}
                            title="Share Usage Data With:"
              />
              <Unknown
                            inverse={true}
                            loading={false}
                            onClick={[Function]}
              >
                            Google Analytics
              </Unknown>
</Styled(styled.div)>,
            <p
              style={
                            Object {
                                          "textAlign": "center",
                                        }
              }
>
              <small>
                            Your usage data is
                             
                            being
                            <span>
                                           anonymously
                            </span>
                             
                            shared with Google Analytics.
              </small>
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.",
            },
            "ref": null,
            "rendered": "Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Analog.Cafe will store your preferences in your browser’s",
                " ",
                <em>
                  LocalStorage
</em>,
                " (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Analog.Cafe will store your preferences in your browser’s",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LocalStorage",
                },
                "ref": null,
                "rendered": "LocalStorage",
                "type": "em",
              },
              " (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithTheme(Component)
                  buttons={
                                    Array [
                                                      0,
                                                    ]
                  }
                  noStar={true}
                  stubborn={true}
                  title="Share Usage Data With:"
/>,
                <Unknown
                  inverse={true}
                  loading={false}
                  onClick={[Function]}
>
                  Google Analytics
</Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttons": Array [
                    0,
                  ],
                  "noStar": true,
                  "stubborn": true,
                  "title": "Share Usage Data With:",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Google Analytics",
                  "inverse": true,
                  "loading": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Google Analytics",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <small>
                Your usage data is
                 
                being
                <span>
                                 anonymously
                </span>
                 
                shared with Google Analytics.
</small>,
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Your usage data is",
                  false,
                  " ",
                  "being",
                  <span>
                     anonymously
</span>,
                  " ",
                  "shared with Google Analytics.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Your usage data is",
                false,
                " ",
                "being",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " anonymously",
                  },
                  "ref": null,
                  "rendered": " anonymously",
                  "type": "span",
                },
                " ",
                "shared with Google Analytics.",
              ],
              "type": "small",
            },
            "type": "p",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            pageTitle="Privacy Settings"
/>,
          <styled.section>
            <p>
                        Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.
            </p>
            <p>
                        Analog.Cafe will store your preferences in your browser’s
                         
                        <em>
                                    LocalStorage
                        </em>
                         (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.
            </p>
            <Styled(styled.div)>
                        <WithTheme(Component)
                                    buttons={
                                                Array [
                                                            0,
                                                          ]
                                    }
                                    noStar={true}
                                    stubborn={true}
                                    title="Share Usage Data With:"
                        />
                        <Unknown
                                    inverse={true}
                                    loading={false}
                                    onClick={[Function]}
                        >
                                    Google Analytics
                        </Unknown>
            </Styled(styled.div)>
            <p
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            >
                        <small>
                                    Your usage data is
                                     
                                    being
                                    <span>
                                                 anonymously
                                    </span>
                                     
                                    shared with Google Analytics.
                        </small>
            </p>
</styled.section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pageTitle": "Privacy Settings",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p>
                Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.
</p>,
              <p>
                Analog.Cafe will store your preferences in your browser’s
                 
                <em>
                                LocalStorage
                </em>
                 (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.
</p>,
              <Styled(styled.div)>
                <WithTheme(Component)
                                buttons={
                                                Array [
                                                                0,
                                                              ]
                                }
                                noStar={true}
                                stubborn={true}
                                title="Share Usage Data With:"
                />
                <Unknown
                                inverse={true}
                                loading={false}
                                onClick={[Function]}
                >
                                Google Analytics
                </Unknown>
</Styled(styled.div)>,
              <p
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
>
                <small>
                                Your usage data is
                                 
                                being
                                <span>
                                                 anonymously
                                </span>
                                 
                                shared with Google Analytics.
                </small>
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.",
              },
              "ref": null,
              "rendered": "Google Analytics is used by Analog.Cafe to anonymously (individual users can not be identified) study how the visitors are using the website and how the website experience could be improved based on those findings.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Analog.Cafe will store your preferences in your browser’s",
                  " ",
                  <em>
                    LocalStorage
</em>,
                  " (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Analog.Cafe will store your preferences in your browser’s",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "LocalStorage",
                  },
                  "ref": null,
                  "rendered": "LocalStorage",
                  "type": "em",
                },
                " (not a cookie) and remember them indefinitely, as long as your browser or you, personally, do not choose to remove that data.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithTheme(Component)
                    buttons={
                                        Array [
                                                            0,
                                                          ]
                    }
                    noStar={true}
                    stubborn={true}
                    title="Share Usage Data With:"
/>,
                  <Unknown
                    inverse={true}
                    loading={false}
                    onClick={[Function]}
>
                    Google Analytics
</Unknown>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttons": Array [
                      0,
                    ],
                    "noStar": true,
                    "stubborn": true,
                    "title": "Share Usage Data With:",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Google Analytics",
                    "inverse": true,
                    "loading": false,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Google Analytics",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <small>
                  Your usage data is
                   
                  being
                  <span>
                                     anonymously
                  </span>
                   
                  shared with Google Analytics.
</small>,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Your usage data is",
                    false,
                    " ",
                    "being",
                    <span>
                       anonymously
</span>,
                    " ",
                    "shared with Google Analytics.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Your usage data is",
                  false,
                  " ",
                  "being",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " anonymously",
                    },
                    "ref": null,
                    "rendered": " anonymously",
                    "type": "span",
                  },
                  " ",
                  "shared with Google Analytics.",
                ],
                "type": "small",
              },
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
